docs/how_to_guide/categorical_predictors.md

```diff
@@ -0,0 +1,163 @@
+
+To make it concrete, let's assume that the stimulus can be the image of a red triangle (Tri) a blue square (Sq), and the context defines the task, for example report either the color (C) or the shape (S) of the stimulus.
+
+```{code-cell} ipython3
+# Example data
+import pandas as pd
+from patsy import dmatrix
+import numpy as np
+
+# Example data 
+data = pd.DataFrame({
+    'stimulus': ['Tri', 'Tri', 'Sq', 'Sq'],  # Stimulus
+    'context': ['C', 'C', 'S', 'S']   # Context
+})
+
+# Spike counts
+counts = np.array([10, 5, 2, 0])
+```
+
+You can use a `patsy` formula to structure the design matrix:
+
+```{code-cell} ipython3
+
+# (a:b) indicates interaction between a and b
+formula = "stimulus + context + stimulus:context"
+```
+
+To construct the design matrix:
+
+```{code-cell} ipython3
+# Generate design matrix
+design_df = dmatrix(formula, data, return_type="dataframe")
+
+design_df
+```
+
+The feature in the design matrix are 1-hot encoded categories.
+Note that `patsy` adds an intercept and drops the reference `Tri`. This is done by design to avoid perfect collinearity (the sum of the `Tri` and the `Sq` column would be equal to the intercept). 
+
+NeMoS GLMs, however, implicitly specify an intercept term, making the first column of this dataframe redundant. Let's drop it.
+
+```{code-cell} ipython3
+design_df.drop(columns=["Intercept"], inplace=True)
+
+design_df
+```
+
+:::{dropdown} Understanding `patsy`'s Output
+:color: info
+:icon: info
+
+In the design matrix:
+- `T.` indicates **treatment coding**, which is used to encode categorical variables. For example, `stimulus[T.Sq]` represents the presence (`1`) or absence (`0`) of the category `Sq` compared to the reference category `Tri`.
+- The reference category (`Tri`) is dropped to avoid collinearity, as including all categories would result in redundancy.
+- Similarly, for `context`, columns `context[C]`, and `context[T.S]` is dropped to avoid collinearity.
+- The interaction term `stimulus[T.Sq]:context[T.S]` represents the combined effect of `stimulus = Sq` and `context = S`. Only one column is needed for the interaction, as the other combinations are implicitly represented by the reference categories (`Tri` and `C`).
+
+See [`patsy` docs](https://patsy.readthedocs.io/en/latest/formulas.html#the-formula-language) for more details. 
+:::
+
+
+### Fit the GLM
+
+We are now ready to fit a GLM model using the 1-hot encoded categories as predictor. The model will learn a different 
+firing rate for each condition. 
+
+
+```{code-cell} ipython3
+import nemos as nmo
+
+# Fit the GLM model
+model = nmo.glm.GLM().fit(design_df, counts)
+```
+:::{note}
+
+You can directly pass a [`pandas.DataFrame`](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.html) to the 
+`fit` method, as long as it can be internally converted to a floating-point array. The conversion is handled automatically, 
+so no additional steps are required on your part.
```

Comment:
```suggestion
You can directly pass a [`pandas.DataFrame`](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.html) to the 
`fit` method, as long as it can be internally converted to a floating-point array (that means, for example, that you can't have a column whose values are strings). The conversion is handled automatically, by calling `design_df.values`,
so no additional steps are required on your part.
```